### PR TITLE
Add test cases to the ReadAtMost function

### PR DESCRIPTION
**Pull request by lunhuijie**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Add UT
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
"NONE"
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/util/tail/tail_test.go [hunk 1]
```diff
@@ -18,10 +18,92 @@ package tail
...
 )
 
+type ReadAtMostReturn struct {
+	data                []byte
+	IsOffsetGreaterZero bool
+	err                 error
+}
+
+func TestReadAtMost(t *testing.T) {
+	//a file "test data A "
```
Comment (lavalamp) [inline pkg/util/tail/tail_test.go:36]: Add spaces between // and text, please.

File: pkg/util/tail/tail_test.go [hunk 2]
```diff
@@ -18,10 +18,92 @@ package tail
...
+	//There will be three results, the file size is enough, the file size is just right, the file size is not enough
+	//Return the expected size data form the end of the file
+
+	//Create the tmp file and write data
+	fileDir, err := ioutil.TempDir("", "file-")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(fileDir)
+	fileOpen, err := os.OpenFile(fileDir+"/testFileA", os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0666)
```
Comment (lavalamp) [inline pkg/util/tail/tail_test.go:48]: Use filepath.Join, as written it will not work on windows.

File: pkg/util/tail/tail_test.go [hunk 3]
```diff
@@ -18,10 +18,92 @@ package tail
...
+	if err != nil {
+		t.Fatalf("Failed to create file %s: %v", fileDir+"/testFileA", err)
+	}
+	defer fileOpen.Close()
+
+	if _, err = fileOpen.WriteString("test data A "); err != nil {
```
Comment (lavalamp) [inline pkg/util/tail/tail_test.go:54]: Can you just write the complete string via ioutil.WriteFile? I don't understand why you would do it in two steps like this.

Comment (lavalamp) [reply pkg/util/tail/tail_test.go:54]: If it's to get the length, you can just declare two constants and call len() on them separately, no?

Comment (lunhuijie) [reply pkg/util/tail/tail_test.go]: ok, thanks

**Comment by yangjunmyfm192085** [comment]
This looks great

**Pull request closed without merging**